r/tests/testthat/test-dplyr-join.R

```diff
@@ -90,9 +90,57 @@ test_that("Error handling", {
     left_tab %>%
       left_join(to_join, by = "not_a_col") %>%
       collect(),
-    "all(names(by) %in% names(x)) is not TRUE",
-    fixed = TRUE
+    "Join column must be present in data"
   )
+  expect_snapshot({
+    (expect_error(
```

Comment:
The extra wrapping of `()`s is to get the error from a whole dplyr chain, right? (at least, that's what I think https://github.com/r-lib/testthat/issues/1471 is saying). 

In these cases, there's not actually a chain, so these should just work (right?)

Reply:
the extra brackets print the output. Without brackets it would look like the example below and we would lose the main reason for using `expect_snapshot()`
<img width="819" alt="image" src="https://user-images.githubusercontent.com/13176361/150107840-d004a6dd-bd8d-4c41-bc86-ea559ed269b4.png">

Reply:
We could use `expect_snapshot(error =  TRUE)` instead of `expect_snapshot()` + `(expect_error())`, but we lose the `Output` heading - something we could live with. See my [6602362](https://github.com/apache/arrow/pull/12179/commits/660236248de9d4d6abaa2f516053eae44260892c) commit.

Reply:
Yup, I like `expect_snapshot(error = TRUE)` and it looks like that's the way that the testthat developers are recommending moving forward too.